docs/API/metadata.md

```diff
@@ -0,0 +1,130 @@
+## Dataset metadata
+
+A dataset is referenced by a formal resource ID, and also has a resource name
+for convenience. The Pbench Server also maintains metadata about each dataset,
+which can help with searching and analysis. Some metadata is modifyable by an
+authenticated user, while other metadata is maintained internally by the server
+and can't be changed. Authenticated users can also add any additional metadata
+that might be of use.
+
+>__NOTE__: right now the ability to search and filter using metadata is
+limited, but our intent is to be able to use any defined metadata value both
+for searches and to filter the results of [datasets/list](V1/list.md).
+
+When a dataset is first processed, the Pbench Server will populate basic
+metadata, including the creation timestamp, the owner of the dataset (the
+username associated with the token given to the Pbench Agent
+`pbench-results-move` command), and the full contents of the dataset's
+`metadata.log` file inside the dataset tarball. These are all accessible
+under the `dataset` metadata key namespace.
+
+The Pbench Server will also calculate a default deletion date for the dataset
+based on the owner's retention policy and the server administrator's retention
+policy along with some other internal management context. The expected deletion
+date is accessible under the `server` metadata key namespace.
+
+Clients can also set arbitrary metadata through the "dashboard" and "user"
+metadata namespaces:
+* The "dashboard" namespace can only be modified by the owner of the dataset,
+and is visible to anyone with read access to the dataset.
+* The "user" namespace is private to each authenticated user, and even if you
+don't own a dataset you can set your own private "user" metadata to help you
+categorize that dataset and to find it again.
+
+Metadata namespaces are hierarchical, and are exposed as nested JSON objects.
+You can address an entire namespace, e.g., `dashboard` or `dataset` and
+retrieve the entire JSON object, or you can address nested objects or values
+using a dotted metadata key path like `dashboard.contact.email` or
+`dataset.metalog.pbench.script`.
+
+For example, given the following hypothetical `user` JSON value:
+
+```json
+{
+    "project": ["OCP", "customer"],
+    "tracker": {"examined": "2022-05-15", "revisit": true},
+    "analysis": {"cpu": "high", "memory": "nominal"}
+}
+```
+
+requesting the metadata `user` (e.g., with `/api/v1/datasets/list?metadata=user`)
+would return the entire JSON value. In addition:
+* `project` would return `["OCP", "customer"]`
+* `user.tracker.examined` would return `"2022-05-15"`
+* `user.analysis` would return `{"cpu": "high", "memory": "nominal"}`
+
+## Metadata namespaces
+
+There are currently four metadata namespaces.
+
+* The `dataset` and `server` namespaces are defined and managed by Pbench.
+* The `dashboard` namespace allows an authenticated client to define an
```

Comment:
Perhaps we should consider a name change from `dashboard` to ... something else?  Perhaps "owner"?

It feels to me like the name "dashboard" is tied to the UI, but this is an API which can be used independent of the Pbench Dashboard.

Reply:
I've been thinking of this since forever, and especially when I changed the `user` namespace. I just haven't gotten around to doing it.

I'd been thinking `client`, but never entirely convinced myself since that's somewhat of an alias for "user" in this context. So `owner` might make sense. Or we could go for something more "meta" like, um, `meta`. Or `global` (which breaks `owner`'s symmetry with `user` to avoid perhaps implying that the values are only visible to the owner?)

Reply:
Naming is hard.